homeassistant/components/hvac/zwave.py

```diff
@@ -0,0 +1,235 @@
+        for value in self._node.get_values(class_id=fan_class_id).values():
+            self._current_operation_state = value.data
+            self._fan_list = list(value.data_items)
+            _LOGGER.debug(('self._fan_list=%s'), self._fan_list)
+            _LOGGER.debug(('self._current_operation_state=%s'),
+                          self._current_operation_state)
+        # Swing mode
+        if self._zxt_120 == 1:
+            for value in self._node.get_values(class_id=0x70).values():
+                if value.command_class == 112 and value.index == 33:
+                    self._current_swing_mode = value.data
+                    self._swing_list = [0, 1]
+                    _LOGGER.debug(('self._swing_list=%s'), self._swing_list)
+
+    @property
+    def should_poll(self):
+        """No polling on ZWave."""
+        return False
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan speed set."""
+        return self._current_operation_state
+
+    @property
+    def fan_list(self):
+        """List of available fan modes."""
+        return self._fan_list
+
+    @property
+    def current_swing_mode(self):
+        """Return the swing mode set."""
+        return self._current_swing_mode
+
+    @property
+    def swing_list(self):
+        """List of available swing modes."""
+        return self._swing_list
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        unit = self._unit
+        if unit == 'C':
+            return TEMP_CELSIUS
+        elif unit == 'F':
+            return TEMP_FAHRENHEIT
+        else:
+            _LOGGER.exception(('unit_of_measurement=%s is not valid'),
+                              unit)
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def current_operation(self):
+        """Return the current operation mode."""
+        return self._current_operation
+
+    @property
+    def operation_list(self):
+        """List of available operation modes."""
+        return self._operation_list
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    def set_temperature(self, temperature):
+        """Set new target temperature."""
+        for value in self._node.get_values(class_id=0x43).values():
+            if value.command_class != 67:
+                continue
+            if self._zxt_120 == 1:
+                # ZXT-120 does not support get setpoint
+                self._target_temperature = temperature
+                if ZXT_120_SET_TEMP.get(self._current_operation) \
+                   != value.index:
+                    continue
+                    _LOGGER.debug(('index=%s, data=%s, temp=%s, op=%s'), value.index, value.data, temperature, self._current_operation)
```

Comment:
I am having some issues here. 
When Heat mode is selected, index 2 and 4 is set.
When Cool mode is selected index 1, 2 and 4 is set.
This won't work ofcourse. Any ideas?